### DISCUSSION TITLE
Little Error in Lesson 10?

### DISCUSSION THREAD

**Discussion by guiri93** [smartcontractkit/full-blockchain-solidity-course-py#648] [Q&A] [answered]
Hello, 
so when I call the function
`get_borrowable_data(lending_pool, account)`
after repaying, it shows still a high amount of borrowed eth.
I think we have to call repay with the following parameters:
`repay_all(amount_dai_to_borrow * (10 ** 18), lending_pool, account).`

**Comment by cromewar** [comment]
Hello @guiri93 nice punctuation, what do you think about this @PatrickAlphaC ?

**Reply by PatrickAlphaC** [reply]
You will have some leftover, our repay function only repays what we have, and you'll have accrued some interest so you'll still have some to pay back

**Reply by cromewar** [reply]
That's interesting, @guiri93 does this makes sense to you?

**Comment by guiri93** [comment]
Yes, I see the point. However when I run repay with my parameters, in the end the function

get_borrowable_data(lending_pool, account)

returns something like 

"You have 1.78...E-9 worth of ETH borrowed."

This small number is explainable by the interests we paid.
But when I run the wave_borrow.py script from the GitHub, after borrowing, the function 

get_borrowable_data(lending_pool, account)

gives something like

"You have 0.078375000000000003 worth of ETH borrowed."

After repaying the function gives

"You have 0.078342892303195526 worth of ETH borrowed."

So we actually only repaid 0.00003... worth of ETH.
I think this can not be explained by the interests we paid in this short time

**Comment by albert-garreta** [comment, marked as answer]
Hi,

I am having the same problem. I think @guiri93 is right about having to repay `amount_to_borrow`. If I'm not mistaken right now we are repaying 0.1 (`AMOUNT`) DAI but we borrowed 95% of 0.1 ETH worth in DAI

**Reply by cromewar** [reply]
They have good points and I'm not really sure how to respond, could you give me hand please @PatrickAlphaC ?

**Reply by PatrickAlphaC** [reply]
Could you show the outputs of your scripts to get to what you're seeing?

**Reply by guiri93** [reply]
Sorry it took me so long.

This is the script from the GitHub repo:

(python397) user@Air-de-Jan AAVE_Brownie % brownie run scripts/solution_script.py 
Brownie v1.17.2 - Python development framework for Ethereum

AaveBrownieProject is the active project.

Launching 'ganache-cli --accounts 10 --hardfork istanbul --fork https://mainnet.infura.io/v3/ddfc1c9cbbd943369147bfe060fd5c0d --gasLimit 12000000 --mnemonic brownie --port 8545 --chainId 1'...

Running 'scripts/solution_script.py::main'...
Transaction sent: 0x76e785736e8b1c4f8eb0f1342753c3dae770570968d25d12f13eeb8ea5f8379f
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 2
  IWeth.deposit confirmed   Block: 14023945   Gas used: 43738 (0.36%)

  IWeth.deposit confirmed   Block: 14023945   Gas used: 43738 (0.36%)

Received 0.1 Weth
Approving ERC20 token...
Transaction sent: 0x4d35baf357e7911808df9a44748ed7d6daefc476ed13620711c6b68d603a3056
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 3
  IERC20.approve confirmed   Block: 14023946   Gas used: 43952 (0.37%)

  IERC20.approve confirmed   Block: 14023946   Gas used: 43952 (0.37%)

Approved!
Depositing...
Transaction sent: 0xc81bc9e363efd631b5a14a13ee2a3d07bde6856eacb10775db54f63d5a635c6e
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 4
  ILendingPool.deposit confirmed   Block: 14023947   Gas used: 250114 (2.08%)

  ILendingPool.deposit confirmed   Block: 14023947   Gas used: 250114 (2.08%)

Deposited!
You have 0.1 worth of ETH deposited.
You have 0 worth of ETH borrowed.
You can borrow 0.0825 worth of ETH.
Let's borrow!
The DAI/ETH price is 0.00030872052743197
We are going to borrow 253.87038773205904 DAI
Transaction sent: 0xe06b0a231b07ac451f574ce17a7257e7394fa7df8f12f8802509758e51018ccb
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 5
  ILendingPool.borrow confirmed   Block: 14023948   Gas used: 369970 (3.08%)

  ILendingPool.borrow confirmed   Block: 14023948   Gas used: 369970 (3.08%)

We borrowed some DAI!
You have 0.100000000002822056 worth of ETH deposited.
You have 0.078374999999999993 worth of ETH borrowed.
You can borrow 0.004125000002328203 worth of ETH.
Approving ERC20 token...
Transaction sent: 0xa3517cb25b4abd78cb60b64c51f6562a7065aa14960b66aef115a6eecbda8e29
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 6
  IERC20.approve confirmed   Block: 14023949   Gas used: 44106 (0.37%)

  IERC20.approve confirmed   Block: 14023949   Gas used: 44106 (0.37%)

Approved!
Transaction sent: 0x1fc6842ecdabf92cfde038aa72ed86c564204f92de3777494338f5ec9eddec34
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 7
  ILendingPool.repay confirmed   Block: 14023950   Gas used: 261245 (2.18%)

  ILendingPool.repay confirmed   Block: 14023950   Gas used: 261245 (2.18%)

Repaid!
You just deposited, borrowed, and repayed with Aave, Brownie, and Chainlink!
You have 0.100000000004389865 worth of ETH deposited.
You have 0.078344129431760665 worth of ETH borrowed.
You can borrow 0.004155870571860974 worth of ETH.
Terminating local RPC client...


And this is the output when I change the parameters of the repay function:


(python397)user@Air-de-Jan AAVE_Brownie % brownie run scripts/aave_borrow.py    
Brownie v1.17.2 - Python development framework for Ethereum

AaveBrownieProject is the active project.

Launching 'ganache-cli --accounts 10 --hardfork istanbul --fork https://mainnet.infura.io/v3/ddfc1c9cbbd943369147bfe060fd5c0d --gasLimit 12000000 --mnemonic brownie --port 8545 --chainId 1'...

Running 'scripts/aave_borrow.py::main'...
Transaction sent: 0x76e785736e8b1c4f8eb0f1342753c3dae770570968d25d12f13eeb8ea5f8379f
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 2
  IWeth.deposit confirmed   Block: 14023962   Gas used: 43738 (0.36%)

  IWeth.deposit confirmed   Block: 14023962   Gas used: 43738 (0.36%)

Received 0.1 Weth
approving erc20 token
Transaction sent: 0x4d35baf357e7911808df9a44748ed7d6daefc476ed13620711c6b68d603a3056
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 3
  IERC20.approve confirmed   Block: 14023963   Gas used: 43952 (0.37%)

  IERC20.approve confirmed   Block: 14023963   Gas used: 43952 (0.37%)

approved
depositing
Transaction sent: 0xc81bc9e363efd631b5a14a13ee2a3d07bde6856eacb10775db54f63d5a635c6e
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 4
  ILendingPool.deposit confirmed   Block: 14023964   Gas used: 250114 (2.08%)

  ILendingPool.deposit confirmed   Block: 14023964   Gas used: 250114 (2.08%)

deposeted
You have 0.1 worth of eth deposited
You have 0 worth of eth borrowed
You can borrow 0.0825 worth of eth 
lets borrow Dai
dai_eth_pride is 0.00030872052743197
we are going to borrow 187.06239096046454 dai.
Transaction sent: 0xa68d02819416f24a52d04189a0f4abf331f03fa1b7cda9c74277b8f0385085dc
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 5
  ILendingPool.borrow confirmed   Block: 14023965   Gas used: 369982 (3.08%)

  ILendingPool.borrow confirmed   Block: 14023965   Gas used: 369982 (3.08%)

borrowed_some_dai
You have 0.100000000002508494 worth of eth deposited
You have 0.057749999999999989 worth of eth borrowed
You can borrow 0.024750000002069519 worth of eth 
amount: 100000000000000000
amount x 100: 10000000000000000000
amount_dai_to_borrow: 187.06239096046454
approving erc20 token
Transaction sent: 0x9fa3a52566bb52431185158108486a7253eff82f15fb3455290046dd3ee9d33c
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 6
  IERC20.approve confirmed   Block: 14023966   Gas used: 44142 (0.37%)

  IERC20.approve confirmed   Block: 14023966   Gas used: 44142 (0.37%)

approved
Transaction sent: 0x87d5c371cac05f8d96c81691aca951946efa8485ef1717a884440e80bfca8e9f
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 7
  ILendingPool.repay confirmed   Block: 14023967   Gas used: 261281 (2.18%)

  ILendingPool.repay confirmed   Block: 14023967   Gas used: 261281 (2.18%)

repayed
you just, deposited, borrowed and repaid with aave and chainlink
You have 0.100000000004076303 worth of eth deposited
You have 1.093844959E-9 worth of eth borrowed
You can borrow 0.082499998909517991 worth of eth 
Terminating local RPC client...

**Reply by PatrickAlphaC** [reply]
I just tried this... it looks like you're right. I made a big oopsie... Thanks for catching!!! I'll make a PR to fix...

**Reply by PatrickAlphaC** [reply]
I have updated here: https://github.com/PatrickAlphaC/aave_brownie_py_freecode/commit/ff488acaec1da4ce143f3ff3763f43bb446e9550 

And am making a PR to the issues doc for this. Looks like some others tried to tell me and I lazily didn't look deep enough.

Thank you both so much!! Sorry for the confusion :( 

Appreciate it!

**Reply by PatrickAlphaC** [reply]
Updated in this repo here: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/commit/d64d413242c334147f0a421d9d7215d289882403

Thanks again!!
